app/src/main/java/com/onewelcome/showcaseapp/feature/userderegistration/UserDeregistrationScreen.kt

```diff
@@ -0,0 +1,164 @@
+package com.onewelcome.showcaseapp.feature.userderegistration
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.material3.Button
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavController
+import com.github.michaelbull.result.Result
+import com.github.michaelbull.result.onFailure
+import com.github.michaelbull.result.onSuccess
+import com.onegini.mobile.sdk.android.handlers.error.OneginiError
+import com.onegini.mobile.sdk.android.model.entity.UserProfile
+import com.onewelcome.core.components.SdkFeatureScreen
+import com.onewelcome.core.components.ShowcaseDropdownMenu
+import com.onewelcome.core.components.ShowcaseFeatureDescription
+import com.onewelcome.core.components.ShowcaseStatusCard
+import com.onewelcome.core.theme.Dimensions
+import com.onewelcome.core.theme.separateItemsWithComa
+import com.onewelcome.core.util.Constants
+import com.onewelcome.showcaseapp.R
+import com.onewelcome.showcaseapp.feature.userderegistration.UserDeregistrationViewModel.State
+import com.onewelcome.showcaseapp.feature.userderegistration.UserDeregistrationViewModel.UiEvent
+
+@Composable
+fun UserDeregistrationScreen(
+  navController: NavController,
+  viewModel: UserDeregistrationViewModel = hiltViewModel()
+) {
+  UserDeregistrationScreenContent(
+    viewModel.uiState,
+    { navController.popBackStack() },
+    { viewModel.onEvent(it) }
+  )
+}
+
+@Composable
+private fun UserDeregistrationScreenContent(
+  uiState: State,
+  onNavigateBack: () -> Unit,
+  onEvent: (UiEvent) -> Unit
+) {
+  LaunchedEffect(Unit) {
+    onEvent.invoke(UiEvent.LoadInitialData)
```

Comment:
why not use `init {}` in viewModel?

Reply:
Hmm I prefer the init {} because it reduce the boilerplate, no need to call the viewmodel from the ui on init. 

From what I read `LaunchedEffect` is useful when you want to reload the data when you come back to the screen, we do not have such a case here though that's true it's easier to test. 

I'm fine with both solutions, though personally I prefer the init when there's no need to reload data on coming back to the screen.  

I think we shouldn't keep two ways of loading initial data on the screen, we should use just one, so need to decide here.